packages/hw-app-ada/src/utils.js

```diff
@@ -0,0 +1,221 @@
+//@flow
+import basex from "base-x";
+
+const BASE58_ALPHABET =
+  "123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz";
+const bs58 = basex(BASE58_ALPHABET);
+
+// We use bs10 as an easy way to parse/encode amount strings
+const bs10 = basex("0123456789");
+
+// Max supply in lovelace
+const MAX_LOVELACE_SUPPLY_STR = ["45", "000", "000", "000", "000000"].join("");
+
+export const Precondition = {
+  // Generic check
+  check: (cond: boolean) => {
+    if (!cond) throw "Precondition failed";
+  },
+  // Basic types
+  checkIsString: (data: any) => {
+    Precondition.check(typeof data === "string");
+  },
+  checkIsInteger: (data: any) => {
+    Precondition.check(Number.isInteger(data));
+  },
+  checkIsArray: (data: any) => {
+    Precondition.check(Array.isArray(data));
+  },
+  checkIsBuffer: (data: any) => {
+    Precondition.check(Buffer.isBuffer(data));
+  },
+
+  // Extended checks
+  checkIsUint32: (data: any) => {
+    Precondition.checkIsInteger(data);
+    Precondition.check(data >= 0);
+    Precondition.check(data <= 4294967295);
```

Comment:
what about Math.pow(2, 32)?

Reply:
I am not really sure whether Math.pow works exactly over integers ...